terraso_backend/apps/core/management/commands/backup.py

```diff
@@ -0,0 +1,83 @@
+import json
+import os
+import traceback
+from datetime import datetime
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from django.core import management
+from django.core.management.base import BaseCommand
+from django.db import connection
+
+from ._backup_storage import S3BackupStorage
+
+
+class Command(BaseCommand):
+    """Backs up the application database to JSON files."""
+
+    help = "Back up database to JSON files"
+
+    def add_arguments(self, parser):
+        parser.add_argument("--s3", action="store_true", help="Store backup in S3 bucket")
+
+    @staticmethod
+    def _query_migration_versions():
+        """Get the migrations applied on the current DB."""
+        with connection.cursor() as cursor:
+            cursor.execute(
+                """SELECT DISTINCT ON (app) app, name
+                   FROM django_migrations
+                    ORDER BY app, name DESC"""
+            )
+            rows = cursor.fetchall()
+        return dict(rows)
+
+    @staticmethod
+    def _save_migration_versions(filename, migration_versions):
+        """Write the migration versions to a JSON file. For now."""
```

Comment:
I think I originally wanted to write the migrations JSON directly to the backup JSON file. I realized later that this would be way more trouble than it's worth. I'll remove the comment.